File: timeTest.java
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class timeTest {

    public static void main(String[] args) {
        int size = 1000000000;
        int[] data = new int[size];
        for (int i = 0; i < size; i++) {
            data[i] = 1;
        }

        int threads = 5;
        ExecutorService executor = Executors.newFixedThreadPool(threads);
        long[] results = new long[threads];
        int chunkSize = size / threads;

        long startTime = System.nanoTime();
        for (int i = 0; i < threads; i++) {
            final int index = i;
            final int start = index * chunkSize;
            final int end = (index == threads - 1) ? size : start + chunkSize;

            executor.execute(() -> {
                long sum = 0;
                for (int j = start; j < end; j++) {
                    sum += data[j];
                }
                results[index] = sum;
            });
        }

        executor.shutdown();
        while (!executor.isTerminated()) {}

        long sum = 0;
        for (long s : results) {
            sum += s;
        }
        long endTime = System.nanoTime();

        System.out.println("Threads: " + threads);
        System.out.println("Sum paralel: " + sum);
        System.out.println("Time paralel: " + (endTime - startTime) / 1000000 + " ms");

        startTime = System.nanoTime();
        sum = 0;
        for (int i = 0; i < size; i++) {
            sum += data[i];
        }
        endTime = System.nanoTime();

        System.out.println("Sum no parallel: " + sum);
        System.out.println("Time no parallel: " + (endTime - startTime) / 1000000 + " ms");
    }
}

File: timeTest.go
package main

import (
	"fmt"
	"runtime"
	"sync"
	"time"
)

func main() {
	threads := 1
	runtime.GOMAXPROCS(threads)
	size := 1000000000
	data := make([]int, size)
	for i := 0; i < size; i++ {
		data[i] = 1
	}

	results := make([]int64, threads)
	chunkSize := size / threads
	startTime := time.Now()
	var wg sync.WaitGroup

	for i := 0; i < threads; i++ {
		wg.Add(1)
		start := i * chunkSize
		end := (i + 1) * chunkSize
		if i == threads-1 {
			end = size
		}
		go func(start int, end int, index int) {
			defer wg.Done()
			sum := int64(0)
			for j := start; j < end; j++ {
				sum += int64(data[j])
			}
			results[index] = sum
		}(start, end, i)
	}

	wg.Wait()
	var sum int64
	for _, result := range results {
		sum += result
	}
	endTime := time.Now()

	fmt.Printf("Threads: %d\n", threads)
	fmt.Printf("Sum: %d\n", sum)
	fmt.Printf("Time: %v\n", endTime.Sub(startTime))
}
